coalib/output/ConsoleInteraction.py

```diff
@@ -164,35 +158,48 @@ def print_lines(console_printer,
     :param sourcerange:     The SourceRange object referring to the related
                             lines to print.
     """
+
     for i in range(sourcerange.start.line, sourcerange.end.line + 1):
         console_printer.print(format_lines(lines='', line_nr=i),
                               color=FILE_LINES_COLOR,
                               end='')
+
         line = file_dict[sourcerange.file][i - 1].rstrip("\n")
-        tab_width = int(section.get('tab_width', 4))
-        s = SpacingHelper(tab_width)
-        tab_dict = dict(s.yield_tab_lengths(line))
+        # Get suitable lexer depending on filename's extension and if not found
+        # fallback to TextLexer().
+        try:
+            lexer = get_lexer_for_filename(sourcerange.file)
+        except ClassNotFound:
+            lexer = TextLexer()
+        lexer.add_filter(VisibleWhitespaceFilter(
+                             spaces="•", tabs=True,
+                             tabsize=SpacingHelper.DEFAULT_TAB_WIDTH))
+        # highlight() combines lexer and formatter to output a ``str``
+        # object.
```

Comment:
how about `term_highlight = lambda string: highlight(string, TerminalTrueColorFormatter())[:-1]` to reduce the duplication?